zipkin/src/main/java/com/linecorp/armeria/client/tracing/HttpTracingClient.java

```diff
@@ -100,7 +105,8 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
         final String method = ctx.method().name();
         span.kind(Kind.CLIENT).name(method).start();
 
-        SpanContextUtil.setupContext(SPAN_IN_THREAD, ctx, span, tracer);
+        // Ensure the trace context propagates to children
+        ctx.onChild((oldCtx, newCtx) -> newCtx.attr(TRACE_CONTEXT_KEY).set(oldCtx.attr(TRACE_CONTEXT_KEY).get()));
```

Comment:
@trustin this is the thing that sets up propagation.. done here and also the server